### PR TITLE
Update to kotlinx.coroutines 1.6.0

### PR DESCRIPTION
**Pull request by alexvanyo**
Updates AlwaysOn to kotlinx.coroutines 1.6.0, and also updates the tests to use the new kotlinx.coroutines test API.

### CODE REVIEW

File: AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt [hunk 1]
```diff
@@ -99,10 +100,12 @@ class MainActivityTests {
             )
         onView(withId(R.id.state)).check(matches(withText(context.getString(R.string.mode_active_label))))
         onView(withId(R.id.draw_count)).check(matches(withText(context.getString(R.string.draw_count_label, 1))))
+
+        scenario.moveToState(Lifecycle.State.STARTED)
```
Comment (alexvanyo) [inline AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt:104]: If we don't pause the activity, then `runTest` fails with uncompleted work. This makes sense: we are always enqueuing an infinite amount of work, so we need to cancel it before finishing `runTest` by pausing the activity.

**Comment by alexvanyo** [comment]
Hi @JoseAlcerreca and @manuelvicnt , I updated a simple sample here to use the new kotlinx-coroutines-test APIs. This was a pretty straightforward change here, since I wanted fine control over timing due to the sample under test.

File: AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt (hunk 1 above)
Comment (codingjeremy) [reply AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt:104]: I'm confused, should there be some call to pause it? I'm new to this API though, so I might be missing something. (If so, ignore this.) :)

Also, should we use your comment as a comment in the code?

**Review by codingjeremy — APPROVED** [review]
Just one comment to your comment.

File: AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt (hunk 1 above)
Comment (alexvanyo) [reply AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt:104]: `moveToState` `STARTED` will move from `RESUMED` to `STARTED`, thereby pausing it. I'll add my comment inline to make that clear!

Comment (alexvanyo) [reply AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt:105]: Updated!

Comment (codingjeremy) [reply AlwaysOnKotlin/Wearable/src/androidTest/java/com/example/android/wearable/wear/alwayson/MainActivityTests.kt:105]: Perfect, thanks!

**Pull request merged**